types/parameterized_types.go

```diff
@@ -0,0 +1,109 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type IntegerParam struct {
+	Name string
+}
+
+func (m IntegerParam) Equals(o IntegerParam) bool {
+	return m == o
+}
+
+func (p IntegerParam) ToProto() *proto.ParameterizedType_IntegerParameter {
+	panic("not implemented")
+}
+
+func (m *IntegerParam) String() string {
+	return m.Name
+}
+
+type ParameterizedTypeSingleIntegerParam[T VarCharType | FixedCharType | FixedBinaryType | PrecisionTimestampType | PrecisionTimestampTzType] struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	IntegerOption    IntegerParam
+}
+
+func (m *ParameterizedTypeSingleIntegerParam[T]) WithIntegerOption(integerOption IntegerParam) ParameterizedSingleIntegerType {
+	m.IntegerOption = integerOption
+	return m
+}
+
+func (*ParameterizedTypeSingleIntegerParam[T]) isRootRef() {}
+func (m *ParameterizedTypeSingleIntegerParam[T]) WithNullability(n Nullability) Type {
+	m.Nullability = n
+	return m
+}
+
+func (m *ParameterizedTypeSingleIntegerParam[T]) GetType() Type               { return m }
+func (m *ParameterizedTypeSingleIntegerParam[T]) GetNullability() Nullability { return m.Nullability }
+func (m *ParameterizedTypeSingleIntegerParam[T]) GetTypeVariationReference() uint32 {
+	return m.TypeVariationRef
+}
+func (m *ParameterizedTypeSingleIntegerParam[T]) Equals(rhs Type) bool {
+	if o, ok := rhs.(*ParameterizedTypeSingleIntegerParam[T]); ok {
+		return *o == *m
+	}
+	return false
+}
+
+func (*ParameterizedTypeSingleIntegerParam[T]) ToProtoFuncArg() *proto.FunctionArgument {
+	// parameterized type are never on wire so to proto is not supported
+	panic("not supported")
+}
+
+func (m *ParameterizedTypeSingleIntegerParam[T]) ShortString() string {
+	switch any(m).(type) {
+	case *ParameterizedVarCharType:
+		t := &VarCharType{}
+		return t.ShortString()
+	case *ParameterizedFixedCharType:
+		t := &FixedCharType{}
+		return t.ShortString()
+	case *ParameterizedFixedBinaryType:
+		t := &FixedBinaryType{}
+		return t.ShortString()
+	case *ParameterizedPrecisionTimestampType:
+		t := &PrecisionTimestampType{}
+		return t.ShortString()
+	case *ParameterizedPrecisionTimestampTzType:
+		t := &PrecisionTimestampTzType{}
+		return t.ShortString()
+	default:
+		panic("unknown type")
+	}
```

Comment:
Same comment as before, it doesn't need to be pointers you can just use the type without needing the `&`

Reply:
fixed it